renderer/dyna_gl.h

```diff
@@ -36,213 +38,72 @@
 #define GLFUNCCALL
 #endif
 
-// ryan's adds. 04/18/2000.
-//  There was a bunch of stuff that was, I guess, not supported by
-//  Mesa when Outrage did the original port. So buttloads of code
-//  involving glActiveTextureARB() were #ifdef'd to only WIN32.
-//  To make use of the extension more generic, platforms that can handle
-//  it (Linux and Win32, as far as I care), define
-//  _USE_OGL_ACTIVE_TEXTURES, and check that define instead of WIN32 when
-//  using the extension.
-#if (!defined(_USE_OGL_ACTIVE_TEXTURES))
-#if (defined(WIN32))
-#define _USE_OGL_ACTIVE_TEXTURES
-#endif
-#endif
-// end ryan's adds.
-
-#ifdef DECLARE_OPENGL
-#define DYNAEXTERN(_ty, _name) _ty _name = NULL
-#else
-#define DYNAEXTERN(_ty, _name) extern _ty _name;
+inline void CheckError() {
+#if defined(_DEBUG)
```

Comment:
Why error checking only on debug? If there runtime error on release build, what happens?

Reply:
Speed, theoretically. I'm a big fan of "fail early, fail loud" so if there's consensus that we should error-check on Release builds as well, I'd be all for it.

To your second question of "what happens if we ignore the error" - that's `main`'s behavior today. Very few GL calls are error-checked, so errors typically just go ignored. OpenGL appears to ignore failed calls, for the most part.

Reply:
Can do! Just to be clear, though: That will cause the Release build to be much crashier. Right now, GL errors are ignored unless they are explicitly checked-for (there are *some* places that check and then `Int3`). Any ignored GL error just ends up causing a graphics degradation, at most. If I turn this on for the Release builds, it will crash on *any* GL error.

This is something I'd realistically use telemetry for, if we were ok collecting it.

Reply:
@winterheart the more I think about it, the more I actually reverse my initial position: I think we should _not_, for now, enforce these error-checks on Release. We don't have (to my knowledge) a good library capable of generating a stacktrace to print for the user, and even if we did, on Mac and Windows (likely the two most common platforms) there isn't even a console to print the error to.

I have some ideas about how to build an error-reporting mechanism that will be palatable to even the most fanatically private users, and we could tie this in to that if/when we do it. But until then, I think turning OpenGL errors into fatal errors would be worse rather than better.

Reply:
OK, but we still need same behavior both for Debug and Release builds. Let's just replace throwing with verbose logging and unify this block for all environments.

Reply:
I'm curious why we need identical behavior between Debug and Release - that's an unusual requirement. In my experience it's very common for Debug builds to include extra assertions and error checking, with those same checks compiled out for the Release version.

CMake has a RelWithDebugInfo.. is that more what you're wanting?

Reply:
On further thought, this PR is not the forum to debate that product requirement either way. I'll gate this behind a compile-flag setting instead if a Debug check and then open an issue to discuss the higher-level issue.

Reply:
https://github.com/DescentDevelopers/Descent3/issues/528 opened, and https://github.com/DescentDevelopers/Descent3/pull/527/commits/e455ed5fa58615b4d39c9eefae74a85295b5a434 added to gate this for now.